client/lib/js/utils/cryptoconditions/filter.js

```diff
@@ -0,0 +1,24 @@
+
+const filterByType = ({ condition, typeId, maxDepth }) => {
+    let res = [];
+    if (condition.hasOwnProperty('type_id') && condition.type_id === typeId) {
+        res.push(condition);
+    }
+
+    const nextMaxDepth = (maxDepth === parseInt(maxDepth, 10)) ? maxDepth - 1 : null;
```

Comment:
Good point. How about

```
if (condition.hasOwnProperty('subfulfillments') && (maxDepth || maxDepth == null)) {
    ...
    res = res.concat(filterByType({
        maxDepth: maxDepth && maxDepth - 1
    });
}
```

This assumes that `maxDepth` will always be a number, null, or undefined. If it's not one of those, it should probably throw an error using `maxDepth != null && typeof maxDepth !== 'number'`?